### PR TITLE
Adding Type Annotations

### PR DESCRIPTION
**Pull request by ssd71**
For issue #9 
I did some trivial type annotations but I can't seem to be getting any further. Can you please review these changes? In addition to these changes, I also attempted to add type annotations to arguments like config, terminalreporter, report by importing the required modules from _pytest, but it didn't stand with tox. What do you suggest?

EDIT: referenced issue

### CODE REVIEW

File: src/pytest_md/plugin.py [hunk 1]
```diff
@@ -149,13 +150,13 @@ def create_summary(self):
...
 
-    def create_results(self):
+    def create_results(self) -> str:
         """Create results for the individual tests for the Markdown report."""
 
         outcomes = {}
 
         for outcome in (o for o in Outcome if o in self.reports):
-            reports_by_file = collections.defaultdict(list)
+            reports_by_file: Dict[Outcome, List] = collections.defaultdict(list)
```
Comment (hackebrot) [inline src/pytest_md/plugin.py]: We group lists of reports by test file here:

```suggestion
            reports_by_file: Dict[str, List] = collections.defaultdict(list)
```

**Review by hackebrot — CHANGES_REQUESTED** [review]
Thanks for working on this @ssd71! 📝 I added a suggestion to your pull-request.

I do not recommend adding any annotations to variables that we don't have control over.

**Comment by ssd71** [comment]
Could you help me judge where I should put in type annotations?

**Comment by hackebrot** [comment]
Hi @ssd71! 👋 

We can merge your pull-request as is and if we feel we need more type annotations, we can always add them later since they're optional in Python.

Thanks again for your contribution! 📝

**Review by hackebrot — APPROVED** [review]
LGTM 🚀

**Pull request merged**